libs/components/modals/src/lib/modules/modal/modal.component.spec.ts

```diff
@@ -123,18 +124,68 @@ describe('Modal component', () => {
     return TestBed.inject(Router);
   }
 
-  function openModal<T>(modalType: T, config?: Record<string, any>) {
+  async function testLiveAnnouncer(
+    changeElementAfterLoad = false
+  ): Promise<void> {
+    const liveAnnouncer = TestBed.inject(SkyLiveAnnouncerService);
+    // The live announcer element is created when an announcement is made. Doing this to create it.
+    liveAnnouncer.announce('Creating element');
+    const liveAnnouncerElement = document.querySelector(
+      '.sky-live-announcer-element'
+    );
+    const modalInstance = openModal(ModalTestComponent, undefined, true);
+    const modalDialogElement = getModalDialogElement();
+
+    if (!liveAnnouncerElement) {
+      fail(
+        'Announcer element should have been set when live announcer was injected'
+      );
+      return;
+    }
+
+    expect(modalDialogElement.getAttribute('aria-owns')).toEqual(
+      liveAnnouncerElement.id
+    );
+    await expectAsync(getModalElement()).toBeAccessible();
+
+    if (changeElementAfterLoad) {
+      liveAnnouncer.announcerElementChanged.next({
+        id: 'changedId',
+      } as HTMLElement);
+
+      getApplicationRef().tick();
+
+      expect(modalDialogElement.getAttribute('aria-owns')).toEqual('changedId');
+      await expectAsync(getModalElement()).toBeAccessible();
+    }
+
+    closeModal(modalInstance, true);
+    liveAnnouncer.ngOnDestroy();
+  }
+
+  function openModal<T>(
+    modalType: T,
+    config?: Record<string, any>,
+    async = false
+  ) {
     const modalInstance = getModalService().open(modalType, config);
 
     getApplicationRef().tick();
-    tick();
+
+    if (!async) {
+      tick();
+    }
 
     return modalInstance;
   }
 
-  function closeModal(modalInstance: SkyModalInstance) {
+  function closeModal(modalInstance: SkyModalInstance, async = false) {
     modalInstance.close();
-    tick();
+
+    if (!async) {
+      tick();
```

Comment:
Is this needed with everything being synchronous? It's being called from tests that are not `fakeAsync`.

Reply:
The `tick` existed prior to my changes. However, because of running a `toBeAccessible` when we check `aria-owns` the new tests need to be truly async and so I needed a way to toggle the `tick` to not run in those cases. That is what this code that you are calling out is changing. Removing the `tick` causes 8 pre-existing tests to fail.